app/assets/javascripts/trainer_plugin/ahoy.js

```diff
@@ -0,0 +1,285 @@
+/*
+ * Ahoy.js
+ * Simple, powerful JavaScript analytics
+ * https://github.com/ankane/ahoy.js
+ * v0.1.0
+ * MIT License
+ */
+
+/*jslint browser: true, indent: 2, plusplus: true, vars: true */
+
+(function (window) {
+  "use strict";
+
+  var ahoy = window.ahoy || window.Ahoy || {};
+  var $ = window.jQuery || window.Zepto || window.$;
+  var visitId, visitorId, track;
+  var visitTtl = 4 * 60; // 4 hours
+  var visitorTtl = 2 * 365 * 24 * 60; // 2 years
+  var isReady = false;
+  var queue = [];
+  var canStringify = typeof(JSON) !== "undefined" && typeof(JSON.stringify) !== "undefined";
+  var eventQueue = [];
+  var page = ahoy.page || window.location.pathname;
+  var visitsUrl = ahoy.visitsUrl || "/ahoy/visits"
+  var eventsUrl = ahoy.eventsUrl || "/ahoy/events"
+
+  // cookies
+
+  // http://www.quirksmode.org/js/cookies.html
+  function setCookie(name, value, ttl) {
+    var expires = "";
+    var cookieDomain = "";
+    if (ttl) {
+      var date = new Date();
+      date.setTime(date.getTime() + (ttl * 60 * 1000));
+      expires = "; expires=" + date.toGMTString();
+    }
+    if (ahoy.domain) {
+      cookieDomain = "; domain=" + ahoy.domain;
+    }
+    document.cookie = name + "=" + escape(value) + expires + cookieDomain + "; path=/";
+  }
+
+  function getCookie(name) {
+    var i, c;
+    var nameEQ = name + "=";
+    var ca = document.cookie.split(';');
+    for (i = 0; i < ca.length; i++) {
+      c = ca[i];
+      while (c.charAt(0) === ' ') {
+        c = c.substring(1, c.length);
+      }
+      if (c.indexOf(nameEQ) === 0) {
+        return unescape(c.substring(nameEQ.length, c.length));
+      }
+    }
+    return null;
+  }
+
+  function destroyCookie(name) {
+    setCookie(name, "", -1);
+  }
+
+  function log(message) {
+    if (getCookie("ahoy_debug")) {
+      window.console.log(message);
+    }
+  }
+
+  function setReady() {
+    var callback;
+    while (callback = queue.shift()) {
+      callback();
+    }
+    isReady = true;
+  }
+
+  function ready(callback) {
+    if (isReady) {
+      callback();
+    } else {
+      queue.push(callback);
+    }
+  }
+
+  // http://stackoverflow.com/a/2117523/1177228
+  function generateId() {
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function(c) {
+      var r = Math.random()*16|0, v = c == 'x' ? r : (r&0x3|0x8);
```

Comment:
Unexpected use of '|'.<br>Mixed double and single quotes.<br>Unexpected use of '&'.